pkg/segment/results/mresults/seriesresult.go

```diff
@@ -1076,6 +1078,60 @@ func evaluateStandardVariance(sortedTimeSeries []Entry, ts map[uint32]float64, t
 	return ts
 }
 
+// MAD Over Time
+func evaluateMADOverTime(sortedTimeSeries []Entry, ts map[uint32]float64, timeWindow uint32) map[uint32]float64 {
+	for i := range sortedTimeSeries {
+		// Define the start time of the window for the current entry
+		timeWindowStartTime := sortedTimeSeries[i].downsampledTime
+		if timeWindow < sortedTimeSeries[i].downsampledTime {
+			timeWindowStartTime -= timeWindow
+		} else {
+			// Case where the window calculation might underflow
+			ts[sortedTimeSeries[i].downsampledTime] = 0
+			continue
+		}
+		// Index of the first entry within the time window
+		preIndex := sort.Search(len(sortedTimeSeries), func(j int) bool {
+			return sortedTimeSeries[j].downsampledTime >= timeWindowStartTime
+		})
+
+		if preIndex >= i { // Check if there is no previous data point within the window
+			ts[sortedTimeSeries[i].downsampledTime] = 0
+			continue
+		}
+
+		// All values within the time window
+		values := make([]float64, 0, i-preIndex+1)
+		for j := preIndex; j <= i; j++ {
+			values = append(values, sortedTimeSeries[j].dpVal)
+		}
+
+		// Median of these values
+		median := computeMedian(values)
+		// Absolute deviations from the median
+		deviations := make([]float64, len(values))
+		for k, v := range values {
+			deviations[k] = math.Abs(v - median)
+		}
+
+		// Median of the absolute deviations
+		mad := computeMedian(deviations)
+		// Store the MAD in the result map
+		ts[sortedTimeSeries[i].downsampledTime] = mad
+	}
+	return ts
+}
+
+// Median Calculation
+func computeMedian(values []float64) float64 {
+	sort.Float64s(values)
+	n := len(values)
+	if n%2 == 0 {
+		return (values[n/2-1] + values[n/2]) / 2
+	}
+	return values[n/2]
+}
```

Comment:
I have one doubt. 
Case 1:
In your implementation FindPercentileData():

For input 10, 20 ==> the median (50th percentile) returned will be 20 as it uses quickselect and don't average it.

case 2:
For the original implementation of Median Absolute Deviation (as of internet), calculation is done around the values of array and so,

input 10, 20 will return (10 + 20) / 2 = 15 (not a value of given array).

So would you like me to go with your FindPercentileData() or with the calculation shown in case 2 (as I did previously using a compute median function).

mad_over_time() is not enabled for me in Prometheus might not be free so don't know how it gives result.

Reply:
Good catch. In that case you can continue with your `computeMedian()` for now. We should change `FindPercentileData()` so that it does linear interpolation, but that probably belongs in a separate PR